com.ibm.streamsx.hdfs/impl/java/src/com/ibm/streamsx/hdfs/HDFS2FileSink.java

```diff
@@ -986,7 +1053,7 @@ public void reset(Checkpoint checkpoint) throws Exception {
 		
 		// close current file
 		if (fFileToWrite != null)
-			closeFile();
+			fFileToWrite.close();
```

Comment:
The method closeFile() triggers also the submission of an output tuple with the filename and the size. But this is imo not expected.
The file which is closed during a reset is not a final file. This file may be overwritten later, thus it should not trigger an output tuple.
My feeling is that the probably running timer thread may cause a problem if the file which is opened after the reset has the same name as the interrupted file. So you are right we must stop the tread here.